src/parse/parser.rs

```diff
@@ -1120,16 +1110,18 @@ where
 
         let mut val = None;
         debug!("Parser::parse_long_arg: Does it contain '='...");
+        let matches;
```

Comment:
Any specific reason to move `matches` out of the `if` scope? Doesn't look like it's used anywhere else.

Reply:
Yes, it prevents a "borrowed value does not live long enough" error. This wasn't a problem when `OsStr` was used, because `trim_start_matches` could return a reference.